src/System.Linq/tests/ToHashSetTests.cs

```diff
@@ -16,10 +17,38 @@ private class CustomComparer<T> : IEqualityComparer<T>
             public int GetHashCode(T obj) => EqualityComparer<T>.Default.GetHashCode(obj);
         }
 
+        public class Record
+        {
+            public string Name;
+            public int Score;
+
+            private sealed class RecordComparer : IEqualityComparer<Record>
+            {
+                public bool Equals(Record x, Record y)
+                {
+                    if (ReferenceEquals(x, y)) return true;
+                    if (ReferenceEquals(x, null)) return false;
+                    if (ReferenceEquals(y, null)) return false;
+                    if (x.GetType() != y.GetType()) return false;
+                    return string.Equals(x.Name, y.Name) && x.Score == y.Score;
+                }
+
+                public int GetHashCode(Record obj)
```

Comment:
Ah, that's true. In that case a different nit: You can remove the null-handling and do `obj.Name.GetHashCode() * 397`.